docs/proposals/20190919-machinepool-api.md

```diff
@@ -0,0 +1,419 @@
+of machines to the infrastructure provider. Many of the fields are shared with MachineDeployment due
+to infrastructure provider's desire to enable the management of a set of machines with a single
+configuration.
+
+``` go
+type MachinePoolSpec struct
+```
+
+- **To add**
+  - **ClusterName [required]**
+    - Type: `string`
+    - Description: Name of the Cluster this machine pool belongs to.
+  - **Replicas [optional]**
+    - Type: `*int32`
+    - Description: Number of desired machine instances. Defaults to 1.
+  - **Template [required]**
+    - Type: `MachineTemplateSpec`
+    - Description: Machine Template that describes the configuration of each machine instance in a
+      machine pool.
+  - **Strategy [optional]**
+    - Type: `*MachineDeploymentStrategy`
+    - Description: Strategy to employ in replacing existing machine instances in a machine pool.
+  - **MinReadySeconds [optional]**
+    - Type: `*int32`
+    - Description: Minimum number of seconds for which a newly created machine should be ready.
+  - **ProviderIDs [optional]**
+    - Type: `[]string`
+    - Description: ProviderIDs contain a ProviderID for each machine instance that's currently
+      managed by the infrastructure provider belonging to the machine pool.
+
+``` go
+type MachinePoolStatus struct
+```
+
+- **To add**
+  - **NodeRefs [optional]**
+    - Type: `[]corev1.ObjectReference`
+    - Description: NodeRefs contain a NodeRef for each ProviderID in MachinePoolSpec.ProviderIDs.
+  - **Replicas [optional]**
+    - Type: `*int32`
+    - Description: Replicas is the most recent observed number of replicas.
+  - **ReadyReplicas [optional]**
+    - Type: `*int32`
+    - Description: The number of ready replicas for this MachinePool.
+  - **AvailableReplicas [optional]**
+    - Type: `*int32`
+    - Description: The number of available replicas (ready for at least minReadySeconds) for this
+      MachinePool.
+  - **UnavailableReplicas [optional]**
+    - Type: `*int32`
+    - Description: Total number of unavailable machines targeted by this machine pool. This is the
+      total number of machines that are still required for this machine pool to have 100% available
+      capacity. They may either be machines that are running but not yet available or machines that
+      still have not been created.
+  - **ErrorReason [optional]**
+    - Type: `*capierrors.MachinePoolStatusError`
+    - Description: ErrorReason indicates that there is a problem reconciling the state, and will be
+      set to a token value suitable for programmatic interpretation.
+  - **ErrorMessage [optional]**
+    - Type: `*string`
+    - Description: ErrorMessage indicates that there is a problem reconciling the state, and will be
+      set to a descriptive error message.
+  - **Phase [optional]**
+    - Type: `string`
+	- Description: Phase represents the current phase of cluster actuation.
+	  e.g. Pending, Running, Terminating, Failed etc.
+  - **BootstrapReady [optional]**
```

Comment:
Is anything looking at the value of this field and making logic choices based on it?

Reply:
The usage of this field will be identical to it's use in Machine. Once BootstrapReady is true, we won't try to copy Data from BootStrapConfig to the MachinePool Spec.